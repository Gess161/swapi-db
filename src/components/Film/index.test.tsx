import { BrowserRouter, useNavigate } from 'react-router-dom';
import renderer from 'react-test-renderer';
import FilmComponent from '.';

const TestWrapper = (props: any) => {
    const { episode, title } = props;
    const navigate = useNavigate()
    return <FilmComponent navigate={navigate} episode={episode} title={title} />
}

it('renders film card with title "Test"', () => {
    const tree = renderer
        .create(
            <BrowserRouter>
                <TestWrapper episode={1} title={'Test'} />
            </BrowserRouter>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});