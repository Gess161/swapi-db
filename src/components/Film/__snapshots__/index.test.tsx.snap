// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders film card with title "Test" 1`] = `
<div
  className="sc-bdvvtL huSZAf"
  onClick={[Function]}
>
  <img
    alt="film"
    src="https://starwars-visualguide.com/assets/img/films/4.jpg"
  />
  <p>
     
    Test
     
  </p>
</div>
`;
